openstack_controller/datadog_checks/openstack_controller/openstack_controller.py

```diff
@@ -202,58 +199,49 @@ def get_project_scopes(self, instance):
         project_scopes = copy.deepcopy(instance_scope.project_scopes)
         return project_scopes
 
-    def get_network_stats(self, tags):
+    def collect_networks_metrics(self, tags):
         """
         Collect stats for all reachable networks
         """
-        # FIXME: (aaditya) Check all networks defaults to true
-        # until we can reliably assign agents to networks to monitor
-        if is_affirmative(self.init_config.get('check_all_networks', True)):
-            all_network_ids = set(self.get_network_ids())
-
+        networks = self.get_networks()
+        network_ids = self.init_config.get('network_ids', [])
+        filtered_networks = []
+        if not network_ids:
             # Filter out excluded networks
-            network_ids = [
-                network_id
-                for network_id in all_network_ids
-                if not any([re.match(exclude_id, network_id) for exclude_id in self.exclude_network_id_rules])
+            filtered_networks = [
+                network
+                for network in networks
+                if not any([re.match(exclude_id, network.get('id')) for exclude_id in self.exclude_network_id_rules])
             ]
         else:
-            network_ids = self.init_config.get('network_ids', [])
-
-        if not network_ids:
-            self.warning(
-                "Your check is not configured to monitor any networks.\n"
-                + "Please list `network_ids` under your init_config"
-            )
-
-        for nid in network_ids:
-            self.get_stats_for_single_network(nid, tags)
+            for network in networks:
+                if network.get('id') in network_ids:
```

Comment:
That is what I changed as part of this PR. The call to the /2.0/networks endpoint is being done [here](https://github.com/DataDog/integrations-core/pull/2709/files/64a949fe203f43dbe594ac1f046edc1d9f1f4fb4#diff-2debf5e59434c9628df0f8bbc393ab52R206). This logic is to filter down networks we only care about based on the provided configuration. If you set network_ids then it will only return metrics about listed networks, if you set exclude_network_id_rules then it will use the pattern to exclude them.